src/plugins/WebOfTrust/util/BackgroundJobFactory.java

```diff
@@ -0,0 +1,50 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * Factory for {@link BackgroundJob}s which allows all background jobs constructed using an instance
+ * of this class to be terminated with a single method invocation.
+ *
+ * @author bertm
+ * @see BackgroundJobFactoryBase
+ */
+public interface BackgroundJobFactory {
+    /**
+     * Constructs a new background job using the default parameters. When this background job is
+     * {@link BackgroundJob#terminate() terminated}, the running job will be notified by
+     * interruption of its thread. Hence, the job implementer must take care not to swallow
+     * {@link InterruptedException}.
+     * @param job the job to run
+     * @param name the human-readable name of the job
+     */
+    public BackgroundJob newJob(Runnable job, String name);
+
+    /**
+     * {@link BackgroundJob#terminate() Terminates} all live
+     * {@link BackgroundJob background jobs} created by this factory, but does not wait for
+     * completion.
+     * @see BackgroundJob#terminate()
+     * @see #waitForTerminationOfAll(long)
+     */
+    public void terminateAll();
+
+    /**
+     * Checks whether all live {@link BackgroundJob background jobs} created by this factory are
+     * terminated.
+     * @return {@code true} if all jobs are terminated.
+     */
+    public boolean allTerminated();
+
+    /**
+     * Waits for all live {@link BackgroundJob background jobs} created by this factory to
+     * terminate. Note that this method makes a snapshot of the currently live jobs, so jobs
+     * created after the wait has started will not be waited upon. This method returns when all
```

Comment:
Since WOT doesn't need the factories but can just use the termination code of BackgroundJobs themselves, I don't think this is worth fixing now.
Thus, I have deprecated the termation functions of the factories at commit e667fe08ae446a9a6262156101a0c43f7b7f6bee. I've added a FIXME to make people notice that it might make sense to fix the code to make it usable.

-> Done for now